### PR TITLE
Remove library helpers.

### PR DESCRIPTION
**Pull request by damacus**
- Moves the escape_string into the resources using it
- Everything still left using escape_regexp now uses escape string

---
This change is [Reviewable](https://reviewable.io/reviews/sous-chefs/line-cookbook/78)

### CODE REVIEW

**Comment by kitchen-porter** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          2 Errors
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please add labels to this Pull Request</td>
    </tr>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please include a <a href="https://github.com/sous-chefs/line-cookbook/blob/master/CHANGELOG.md">CHANGELOG</a> entry.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This Pull Request is probably missing tests.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: resources/replace_or_add.rb [hunk 1]
```diff
@@ -53,19 +53,7 @@
...
-        nf = ::File.open(new_resource.path, 'w')
-        nf.puts new_resource.line
-      rescue ENOENT
-        Chef::Log.info('ERROR: Containing directory does not exist for #{nf.class}')
-      ensure
-        nf.close
-      end
-    end
-  end
+
```
Comment (MarkGibbons) [inline resources/replace_or_add.rb]: Creating the new file was a deliberate choice.  We are doing an add.

Comment (damacus) [reply resources/replace_or_add.rb]: Noted reverting that change 👍

File: resources/replace_or_add.rb [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
-property :pattern, String
+property :pattern, [String, Regexp]
 property :line, String
 property :replace_only, [true, false]
 
 resource_name :replace_or_add
 
 action :edit do
-  regex = /#{new_resource.pattern}/
+  regex = new_resource.pattern.is_a?(String) ? /#{new_resource.pattern}/ : new_resource.pattern
```
Comment (MarkGibbons) [inline resources/replace_or_add.rb:9]: nice

Comment (damacus) [reply resources/replace_or_add.rb:9]: I'm just standardising that pattern where I can. If/when we remove it, it'll be the same change rather than guessing who, what, why.

File: resources/add_to_list.rb [hunk 3]
```diff
@@ -25,7 +25,7 @@
 
     regexdelim = []
     new_resource.delim.each do |delim|
-      regexdelim << escape_regex(delim)
+      regexdelim << escape_string(delim)
```
Comment (MarkGibbons) [inline resources/add_to_list.rb]: Would this work here: string = Regexp.escape(delim)
The we could get rid of the escape string methods.
Pretty sure this is what Lamont did in append if no line.

File: resources/delete_from_list.rb [hunk 4]
```diff
@@ -1,12 +1,12 @@
 property :path, String
```
Comment (MarkGibbons) [inline resources/delete_from_list.rb:1]: The tests are failing on delete_from_list.  Those are the most fragile and incomplete tests we have in this cookbook. I can look at the tests and failures tomorrow.  Have to deal with some other stuff the rest of today.

**Comment by damacus** [comment]
Same. I have to drop off now and go do some other stuff. Wouldn't be too bad to get some more unit tests for this too.

File: resources/delete_from_list.rb [hunk 5]
```diff
@@ -35,7 +35,7 @@
       when 1
         case line
         when /#{regexdelim[0]}\s*#{new_resource.entry}/
-          replace_deleted = regexdelim[0] =~ /\s+/ ? regexdelim[0] : ''
+          replace_deleted = regexdelim[0] =~ /\s+/ ? new_resource.delim[0] : ''
```
Comment (MarkGibbons) [inline resources/delete_from_list.rb]: This change fixed the failures. But the check regexdelim[0] =~ /\s+/ probably fails in some edge case. I'll try to come up with some better tests.

**Review by lamont-granquist — DISMISSED** [review]

**Comment by MarkGibbons** [comment]
Added a couple more tests for the delete items from resource list.  Cleaned up the delete items code.

**Review by MarkGibbons — APPROVED** [review]
Looks good to me.

File: resources/delete_from_list.rb [hunk 6]
```diff
@@ -35,10 +35,10 @@
       when 1
         case line
         when /#{regexdelim[0]}\s*#{new_resource.entry}/
-          replace_deleted = regexdelim[0] =~ /\s+/ ? regexdelim[0] : ''
-          line = line.sub(/(#{regexdelim[0]})*\s*#{new_resource.entry}(#{regexdelim[0]})*/, replace_deleted)
-          line = line.chomp
-          line = line.rstrip if regexdelim[0] =~ /\s+/
+          # remove the entry
```
Comment (damacus) [inline resources/delete_from_list.rb:38]: Nice. I'm finding it hard to reason what this resource is doing.
